aws_lambda_builders/workflows/java_maven/DESIGN.md

```diff
@@ -0,0 +1,94 @@
+# Java - Maven Lambda Builder
+
+## Scope
+
+This package enables the creation of a Lambda deployment package for Java
+projects managed using the Maven build tool.
+
+## Challenges
+
+- Java Version compatibility mentioned in the [Gradle Lambda Builder] Design doc.
+
+- Building Multimodule project (out of scope for the current version)
+
+Here `ProjectA` is a a single lambda function, and `ProjectB` is a multimodule
+project where sub modules `lambda1` and `lambda2` are each a lambda
+function. In addition, suppose that `ProjectB/lambda1` has a dependency on its
+sibling module `ProjectB/common`.
+
+**Single-module Project A**
+```
+ProjectA
+├── pom.xml
+├── src
+└── template.yaml
+```
+
+**Multi-module Project B**
+```
+ProjectB
+├── common
+│   └── pom.xml
+├── lambda1
+│   └── pom.xml
+├── lambda2
+│   └── pom.xml
+├── pom.xml
+└── template.yaml
+```
+
+Building Project A is relatively simple since we just need to issue `mvn
+package` and place the built classes and dependency jars within the artifact directory.
+
+Building `ProjectB/lambda1` requires maven to build `lambda1` module from
+the root pom directory and use `--also-make` option to build necessary dependencies 
+(`ProjectB/common`  in this case) first before building `ProjectB/lambda1`. This is because
+maven is not able to find its way back up to the parent `ProjectB` to
+also build `ProjectB/common`. The challenge part here is to find the parent pom directory 
+especially for the projects with multiple level of submodules. Building multi-module project is
+out of scope for the current version.
+
+## Implementation
+
+### Build Workflow
+
+#### Step 1: Copy source project to scratch directory
+
+By default, Maven stores its build-related metadata in a `target`
+directory under the source directory and there is no way to change the output 
+directory from command line. To avoid writing anything under `source_dir`, 
+we copy the source project to scratch directory and build it from there.
+
+#### Step 2: Check Java version and emit warning
+
+Check whether the local JDK version is <= Java 8, and if it is not, emit a
+warning that the built artifact may not run in Lambda unless a) the project is
+properly configured (i.e. using `targetCompatibility`) or b) the project is
```

Comment:
Not sure if there's a config in Maven with the same name, but I think the option is different (it's `target`) https://maven.apache.org/plugins/maven-compiler-plugin/examples/set-compiler-source-and-target.html

Reply:
Yup. fixed